graphql_compiler/schema_generation/sqlalchemy/schema_graph_builder.py

```diff
@@ -0,0 +1,81 @@
+from collections import OrderedDict
+import warnings
+
+from graphql.type import GraphQLBoolean, GraphQLFloat, GraphQLInt, GraphQLString
+import six
+import sqlalchemy.sql.sqltypes as sqltypes
+
+from ...schema import GraphQLDate, GraphQLDateTime, GraphQLDecimal
+from ..schema_graph import PropertyDescriptor, SchemaGraph, VertexType
+
+
+# TODO(pmantica1): Add scalar mapping for the following classes: Interval, and Time.
+# We do not currently plan to add a mapping for JSON and Binary objects.
+SQL_CLASS_TO_GRAPHQL_TYPE = OrderedDict([
+    (sqltypes.String, GraphQLString),
+    (sqltypes.Integer, GraphQLInt),
+    (sqltypes.Float, GraphQLFloat),
+    (sqltypes.Numeric, GraphQLDecimal),
+    (sqltypes.DateTime, GraphQLDateTime),
+    (sqltypes.Date, GraphQLDate),
+    (sqltypes.Boolean, GraphQLBoolean),
+])
+
+
+# TODO(pmantica1): Map foreign keys to edges.
+# TODO(pmantica1): Represent table inheritance in SchemaGraph.
+# TODO(pmantica1): Add option to map tables to EdgeTypes instead of VertexTypes.
+# TODO(pmantica1): Map arrays to GraphQLLists.
+def get_schema_graph_from_sql_alchemy_metadata(sqlalchemy_metadata):
+    """Return the matching SchemaGraph for the SQLAlchemy Metadata object"""
+    _validate_sql_to_graphql_is_toposorted_by_class_inheritance()
+    elements = dict()
+    for table_name, table in six.iteritems(sqlalchemy_metadata.tables):
+        elements[table_name] = _get_vertex_type_from_sqlalchemy_table(table)
+    inheritance_sets = {element_name: {element_name} for element_name in elements}
+    return SchemaGraph(elements, inheritance_sets)
+
+
+def _validate_sql_to_graphql_is_toposorted_by_class_inheritance():
+    """Validate that SQL_SCALAR_CLASS_TO_GRAPHQL_TYPE dict is toposorted by class inheritance."""
+    sql_classes = list(SQL_CLASS_TO_GRAPHQL_TYPE.keys())
+    for i, class_ in enumerate(sql_classes):
+        for other_class_ in sql_classes[i + 1:]:
+            if issubclass(other_class_, class_):
+                raise AssertionError("SQL_CLASS_TO_GRAPHQL_TYPE is not toposorted by class "
+                                     "inheritance. Class {} is a subclass of {} but is appears  "
+                                     "later in the OrderedDict.")
+
+
+def _try_get_graphql_scalar_type(column_name, column_type):
+    """Return the most precise GraphQLScalarType for the SQL datatype class.
```

Comment:
This function has the advantage that we can also map custom types to GraphQLTypes provided the custom types inherit from the supported SQL datatypes.

Reply:
Great -- document that in the docstring, please. Also, whenever your docstring is more than just the one-liner, please remember to include `Args:` and `Returns:` sections.

Reply:
I am going to follow Bojan's suggestion for this code to simplify it.  I am just going to add a direct map from primitive to GraphQLType. One advantage that this approach seemed to have is that it dealt with custom types that inherited from primitives, but the complexity doesn't seem worth it.